src/odemis/acq/milling/tasks.py

```diff
@@ -0,0 +1,132 @@
+"""
+@author: Patrick Cleeve
+
+Copyright © 2025 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+
+### Purpose ###
+
+This module contains structures to define milling tasks and parameters.
+
+"""
+
+from typing import Dict, List
+
+import yaml
+from odemis import model
+from odemis.acq.milling.patterns import (
+    MillingPatternParameters,
+    pattern_generator,
+)
+
+
+class MillingSettings:
+    """Represents milling settings for a single milling task"""
+
+    def __init__(self, current: float, voltage: float, field_of_view: float, mode: str = "Serial", channel: str = "ion", align: bool = True):
+        self.current = model.FloatContinuous(current, unit="A", range=(20e-12, 120e-9))
+        self.voltage = model.FloatContinuous(voltage, unit="V", range=(0, 30e3))
+        self.field_of_view = model.FloatContinuous(field_of_view, unit="m", range=(50e-06, 960e-06))
+        self.mode = model.StringEnumerated(mode, choices=set(["Serial", "Parallel"]))
```

Comment:
Typically, sets can be written explicitly as `{a, b,...}`, which is nicer to read:
```suggestion
        self.mode = model.StringEnumerated(mode, choices={"Serial", "Parallel"})
```